### ISSUE TITLE
Docker installation error running freecodecamp locally

### ISSUE THREAD

**Issue by mdalpozzo** [freeCodeCamp/freeCodeCamp#35754]
**Environment:**
MacOS High Sierra v10.13.6 (13'' 2017)
Node 10.4.0
npm 6.9.0
docker 18.09.2
53.59GB free on 250GB drive

**What happened:**

I am following the [setup instructions here](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-setup-freecodecamp-locally-using-docker.md).  During the [docker installation](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-setup-freecodecamp-locally-using-docker.md#installing-dependencies) when I run `npm run docker:install` there is an error that seems to be related to the chromium install that accompanies puppeteer.

<details>
<summary>
Terminal Error Message
</summary>

```
> @freecodecamp/freecodecamp@0.0.1 docker:install /Users/mdalpozzo/Desktop/Coding/open-source/freeCodeCamp
> docker-compose run --rm freecodecamp npm install


Starting freecodecamp_mailhog_1 ... done
Starting freecodecamp_db_1      ... done


> @freecodecamp/freecodecamp@0.0.1 postinstall /app
> npm run bootstrap




> @freecodecamp/freecodecamp@0.0.1 bootstrap /app
> lerna bootstrap


lerna notice cli v3.11.1
lerna info versioning independent
lerna info Bootstrapping 9 packages
lerna info Installing external dependencies
lerna ERR! npm install exited 1 in '@freecodecamp/curriculum'
lerna ERR! npm install stdout:


> puppeteer@1.11.0 install /app/curriculum/node_modules/puppeteer
> node install.js




lerna ERR! npm install stderr:


ERROR: Failed to download Chromium r609904! Set "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" env variable to skip download.
{ [Error: ENOSPC: no space left on device, write] errno: -28, code: 'ENOSPC', syscall: 'write' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! puppeteer@1.11.0 install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the puppeteer@1.11.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.


npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2019-04-03T21_11_22_211Z-debug.log


lerna ERR! npm install exited 1 in '@freecodecamp/curriculum'
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @freecodecamp/freecodecamp@0.0.1 bootstrap: `lerna bootstrap`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @freecodecamp/freecodecamp@0.0.1 bootstrap script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.


npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2019-04-03T21_11_22_370Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @freecodecamp/freecodecamp@0.0.1 postinstall: `npm run bootstrap`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @freecodecamp/freecodecamp@0.0.1 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.


npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2019-04-03T21_11_22_424Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @freecodecamp/freecodecamp@0.0.1 docker:install: `docker-compose run --rm freecodecamp npm install`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @freecodecamp/freecodecamp@0.0.1 docker:install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.


npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mdalpozzo/.npm/_logs/2019-04-03T21_11_23_193Z-debug.log
```

</details>

I added `PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true` to my `.env` file, but run into the exact same error.

Part of the error says something about not having enough disk space, but I should be fine.  I'll keep experimenting, but any help would be amazing!

**Comment by raisedadead** [comment]
> I added PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true to my .env file, but run into the exact same error.

Adding env vars to the `.env` will not be picked up during npm installs. They are for our code logic only, that is when freeCodeCamp is actually running, or a script that we use the `.env` file in.

npm is unaware of this file during its own usage, i.e. in this case installing puppeteer.

What you want to try and check if you are behind a firewall or a network proxy / vpn that is blocking the download.

Try and install puppeteer in a temp directory, to see it works at all.

```
mkdir ~/temp-puppeteer
cd ~/temp-puppeteer
npm init -y
npm i puppeteer
```

**Comment by mdalpozzo** [comment]
I have no firewall running and no vpn running.

I can successfully download puppeteer (including chromium) in a temporary directory.

**Comment by raisedadead** [comment]
Thanks, that pins the issue with the installation it self, so we are on the right track here.

Can you confirm, how you installed docker itself? Also, how are you starting up the docker CLI.

**Comment by mdalpozzo** [comment]
I've found my issue and have successfully gotten FCC running locally on my mac now.   The problem is with Docker desktop and some container memory leak issues.  It seems that this is a known issue with Docker for mac here is the [open issue thread](https://github.com/docker/for-mac/issues/371).  Basically Docker isn't cleaning up when you delete images or containers and the disk image keeps growing.  Even with a 64GB cap set my docker disk image was over 67GB.  I had to reset/delete the Docker disk image which means also losing all images/containers that are currently stored locally.
